src/main/scala/org/ergoplatform/nodeView/state/UtxoState.scala

```diff
@@ -47,7 +49,7 @@ class UtxoState(override val persistentProver: PersistentBatchAVLProver[Digest32
 
   override val maxRollbackDepth = 10
 
-  override def rollbackTo(version: VersionTag): Try[UtxoState] = {
+  override def rollbackTo(version: VersionTag): Try[UtxoState] = persistentProver.synchronized {
```

Comment:
To fully comply JMM, read access to `persistentProver` should be synchronized as well. This should be true for `rollbackVersions`, `boxById`, `randomBox` methods.

I'd prefer making State an actor to enable actor memory model instead